client/dashboard/app/hooks/use-date-range.ts

```diff
@@ -0,0 +1,80 @@
+import { useRouter } from '@tanstack/react-router';
+import { getUnixTime } from 'date-fns';
+import { useState, useRef, useEffect } from 'react';
+import { getDefaultDateRange, getInitialDateRangeFromSearch } from '../../utils/datetime';
+
+interface UseDateRangeOptions {
+	timezoneString?: string;
+	gmtOffset?: number;
+	autoRefresh?: boolean;
+}
+
+export function useDateRange( {
```

Comment:
I'm not sure, but I don't think we've decided to have global hooks under Dashboard V2 yet. Currently, site-performance is the only one here, and it will be moved to the Performance page once that work starts. cc @Automattic/lego Any thoughts on global hooks?